tls/s2n_record_read_stream.c

```diff
@@ -76,7 +76,6 @@ int s2n_record_parse_stream(
     POSIX_GUARD(s2n_hmac_digest(mac, check_digest, mac_digest_size));
 
     if (s2n_hmac_digest_verify(en.data + payload_length, check_digest, mac_digest_size) < 0) {
-        POSIX_GUARD(s2n_stuffer_wipe(&conn->in));
         POSIX_BAIL(S2N_ERR_BAD_MESSAGE);
```

Comment:
We discussed moving this code higher / always wiping on failure-- do you have an open issue for that?

Reply:
Yeah I wanted to merge a PR to do this before merging this fix to avoid a regression - but if this seems involved I will open an issue for it before merging this.

Reply:
PR for this: https://github.com/aws/s2n-tls/pull/4499